### ISSUE TITLE
GNU extension to C Ternary operator that omits second operand flagged as syntax error by VS

### ISSUE THREAD

**Issue by allanhaywood** [Microsoft/VSLinux#114]
Please use the following bug reporting template to help produce actionable and reproducible issues:

* A brief description
A GNU extension to C allows the second operand to be omitted, and the first operand is implicitly used as the second as well:

a = x ? : y;

The expression is equivalent to

a = x ? x : y;

except that if x is an expression, it is evaluated only once. The difference is significant if evaluating the expression has side effects.

Using the format: a = x ? : y;
in code in visual studio causes it to be marked as invalid syntax ("expected expression"), even though it is valid in linux c code.

* Expected results
Using the format: a = x ? : y;
is recognized by VS

* Actual results (with build output window contents if applicable)
Using the format: a = x ? : y;
in code in visual studio causes it to be marked as invalid syntax ("expected expression"), even though it is valid in linux c code.

* VC++ version, Linux system name and version, GCC/GDB/gdbserver version, CPU arch, etc. If you are using a specific version of Linux on an embedded board, we might need to know about it to get a repro on similar hardware
Microsoft Visual Studio Community 2017 RC
Version 15.0.26206.0 D15REL
Microsoft .NET Framework
Version 4.6.01586
Visual C++ for Linux Development   1.0.6

* Steps required to reproduce the error
Use the format: a = x ? : y;
in some code.

* Required packages and commands to install
NA

**Comment by olivier-haddad** [comment]
Hi,
VS Intellisense doesn't know about specific compilers quirks and extensions but only about the options built-in inside VS.
By the way, you'll more often have VS extensions rejected by gnu.

**Comment by allanhaywood** [comment]
Does the c++ extension modify VS Intellisense? If that is the case, then can it be modified to allow this syntax?

**Comment by olivier-haddad** [comment]
No it doesn't. Intellisense in itself is a large project and I doubt the VS extension team will want to mess with it for every linux compiler out there.
I do however copy my linux include folders to the PC so I don't see the red squiggly lines for linux specific calls. But they will still be interpreted by Intellisense according to its interpretation of a valid syntax.

**Comment by allanhaywood** [comment]
Yeah, I have copied the header files locally to get rid of the red squiggly lines for linux calls, so my files show up mostly clean.

This is just one that happened to remain, and isn't due to missing header files that need to be included, just a syntax difference between windows/linux.

If there is no chance of this being addressed, I'll leave it to the VSLinux team to close this as "won't fix".

**Comment by itodirel** [comment]
I would personally recommend not using non standard lamguage extensions, GCC, Clang or Visual C++ ones, but sometimes they are necessary. 

One way you can fix your code, if you want to use the extension, is to effectively turn off IntelliSense for that line, you can do so like this:

```
#ifndef __INTELLISENSE__
	a = x ? : y;
#endif
```

This will make the line invisible to IntelliSense, but not to build.

**Comment by itodirel** [comment]
Please let us know if you have more questions or it doesn't work for you
